### PR TITLE
Update to list Distros module in the manual

### PR DESCRIPTION
**Pull request by rayman22201**
Update to list Distros module in the manual

I was updating some nim packages that I needed for Mac compatibility, and it took me a long time to find the correct way to write platform specific #ifdef code. I figured it out by looking at other nim projects like the nim-glfw project.

I tracked down the ```macosx``` constant to the ```distros``` module, but I didn't find it easily listed in the docs anywhere.
This is my attempt to add the ```distros``` module to the main docs page to make it easier to find for other people.

I hope my description and placement of the module is correct.

Thanks,
Ray

### CODE REVIEW

**Comment by dom96** [comment]
Hey! Thanks for this PR :)

Definitely should have the ``distros`` module in that list, but I think your description is a bit incorrect. The ``distros`` module doesn't define these ``windows``, ``macosx``, and ``linux`` symbols that are widely used throughout the code. This is sort of defined by the compiler magically :)

We should document that somewhere too.

**Comment by rayman22201** [comment]
Where in the docs should the "*magical compiler defined things*" ™️  section live? 😝 

The distros module is important, but the real *gotcha* for me was those compiler defined symbols. They are really important for cross-platform code, and should have a place in the docs somewhere.

File: doc/lib.rst [hunk 1]
```diff
@@ -64,6 +64,11 @@ Core
 * `cpuinfo <cpuinfo.html>`_
   This module implements procs to determine the number of CPUs / cores.
 
+* `distros <distros.html>`_
+  This module implements the basics for OS distribution ("distro") detection and the OS's native package manager.
+  Its primary purpose is to produce output for Nimble packages, but it also contains the widely used **Distribution** enum
+  that is useful for writing platform specific code. It is imported implicitly by the compiler.
```
Comment (Araq) [inline doc/lib.rst]: It is not imported implicitly by the compiler! You can import it directly, it does not rely on compiler magic to work.

Comment (rayman22201) [reply doc/lib.rst]: My confusion is that the platform enum is defined by compiler magic, but the distros module is not.

I need to separate the two, but I am not sure where to put the documentation about the enum.

**Pull request merged**